react-cors-spa/src/Products.js

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios';
+import { useQuery } from '@tanstack/react-query';
+import { DataGrid } from '@mui/x-data-grid';
+import { ThemeProvider, CssBaseline } from '@mui/material';
+import { useMemo } from "react";
+import { theme } from "./ProductStyles";
+
+function fetchProducts() {
+    return axios.get('http://127.0.0.1:3000/products?limit=5&offset=0');
+}
+
+function Products() {
+    const { isLoading, error, data } = useQuery({
+      queryKey: ['products'],
+      queryFn: fetchProducts,
+    });
+
+    const rows = useMemo(() => 
+        data?.data.map((product) => ({
+            id: product.product_id,
+            name: product.name,
+            description: product.description,
+        })),
+    [data]);
+
+    const columns = [
+        { field: 'id', headerName: 'ID', width: 150},
+        { field: 'name', headerName: 'Name', width: 150 },
+        { field: 'description', headerName: 'Description', width: 250},
+    ];
+
+    return (
+        <ThemeProvider theme={theme} defaultMode="dark">
```

Comment:
The Theme Provider should be really high up in our app because we want to to apply globally.